shared/common-adapters/icon.native.js

```diff
@@ -11,74 +10,47 @@ import {omit} from 'lodash'
 import type {Exact} from '../constants/types/more'
 import type {IconType, Props} from './icon'
 
-class Icon extends Component<void, Exact<Props>, void> {
-  render () {
-    let color = shared.defaultColor(this.props.type)
-    let iconType = shared.typeToIconMapper(this.props.type)
+const Icon = (props: Exact<Props>) => {
+  let iconType = shared.typeToIconMapper(props.type)
 
-    if (!iconType) {
-      console.warn('Null iconType passed')
-      return null
-    }
-
-    color = this.props.style && this.props.style.color || color || (this.props.opacity ? globalColors.lightGrey : globalColors.black_40)
+  if (!iconType) {
+    console.warn('Null iconType passed')
+    return null
+  }
+  if (!iconMeta[iconType]) {
```

Comment:
i moved some lines in this file around so its more consistent. the bailing checks up top